custom/icds_reports/sqldata/exports/dashboard_usage.py

```diff
@@ -183,81 +135,46 @@ def get_excel_data(self):
-                    continue
-                indicator_count = user_indicators[user['username']]
-                user_sql_location_ids = user['assigned_location_ids']
-                if isinstance(user_sql_location_ids, str):
-                    user_sql_location_ids = [user_sql_location_ids]
-
-                for user_sql_location in user_sql_location_ids:
-                    # getting the location type to look up in matrix
-                    if user_sql_location not in self.sql_locations:
-                        continue
-                    location_type_id = self.sql_locations[user_sql_location]
-                    column_index = self.location_types.index(location_type_id)
-                    user_location_row = None
-                    # iterating and getting the db row from matrix
-                    for row in location_matrix:
-                        if row[2 * column_index] == user_sql_location:
-                            user_location_row = copy.deepcopy(row)
-                            break
-                    user_location_type = self.get_location_type_string_from_location_id(location_type_id)
-
-                    if user_location_row is not None:
-                        if user_location_type == 'state':
-                            user_location_row[3] = ''
-                            user_location_row[5] = ''
-                        if user_location_type == 'district':
-                            user_location_row[5] = ''
-
-                        last_login, logged_in_last_week = self.check_if_date_in_last_week(user['last_login'])
-
-                        excel = [serial_count, user_location_row[1], user_location_row[3],
-                                 user_location_row[5], user['username'].split('@')[0], user_location_type,
-                                 self.get_role_from_username(user['username']),
-                                 self.convert_boolean_to_string(self.agg_list[user_sql_location]),
-                                 last_login,
-                                 self.convert_boolean_to_string(logged_in_last_week),
-                                 user_counts[user['username']]]
-                        excel.extend(indicator_count)
-                        excel_rows.append(excel)
+            date = datetime.datetime.now().date()
+            usage_data = None
+            dashboard_filters = {'username__in': usernames}
+            # keep the record in searched - current - month
+            while usage_data is None:
+                usage_data = self.get_data_for_usgae_report(date, dashboard_filters)
+                date -= relativedelta(days=1)
+
+            for record in usage_data:
+                last_activity, activity_in_last_week = self.check_if_date_in_last_week(record.last_activity)
+                district_name = self.get_location_name_from_id(record.district_id)
+                block_name = self.get_location_name_from_id(record.block_id)
```

Comment:
What will happen is the user is district level and it will have 'All' as its value. Same if a user is Staet level User.

Reply:
intially we were getting these values with aggregation level 5. So there was no chance of all. Since we have moved to new implementation, i have changed accordingly

Reply:
What will happen when `record.block_id` is 'All' what will get_location_name_from_id return?